src/coreComponents/mesh/NodeManager.hpp

```diff
@@ -424,63 +408,63 @@ class NodeManager : public ObjectManagerBase
    * @note An error is thrown if the total displacement does not exist
    */
   array2d< real64, nodes::TOTAL_DISPLACEMENT_PERM > & totalDisplacement()
-  { return getReference< array2d< real64, nodes::TOTAL_DISPLACEMENT_PERM > >( viewKeys.totalDisplacement ); }
+  { return getExtrinsicData< extrinsicMeshData::solidMechanics::totalDisplacement >(); }
```

Comment:
I was unusure about what to do with these getters. They are handy and used everywhere in the code but they seem to introduce a dependency on the solvers (that register the total displacement) now that the string is no longer hard-coded in the `NodeManager`. I was going to remove them, but I don't know what other people think.

Reply:
They should've never existed in the first place IMO

Reply:
The pattern we are aiming is to let the "solvers" add the fields they need.
`NodeManager` manages nodes; and as an `ObjectManager` it allows fields registrations on nodes.

We need to shrink the contracts.

For instance, why would we want all the code manipulating the `NodeManager` to be bothered with the `incrementalDisplacement`?

Now be careful if you want to remove them, because that means that everywhere these getters are accessed, it's assumed that the field is there (and therefore registered).
So the patch may not be trivial in the sense where you not only need to change the getter call into and getExtrinsicData call, but also think about the registration. It shouldn't be part of this PR.

Reply:
Done, I removed these accessors from the `NodeManager`, I think it is simpler to do it in this PR.

Reply:
Yes, the issue didn't look as challenging as I first thought, since the registrations are already centralized in some solver function ! Good news!